Dockerfile

```diff
@@ -1,17 +1,45 @@
-from golang:1.7-alpine AS main
+from golang:1.9-alpine AS main
+workdir /go/src/github.com/tonistiigi/copy
 copy . .
-run go build -o /copy -ldflags "-s -w" main.go
+run go build -o /copy -ldflags "-s -w" ./
 
 from lalyos/upx AS upx
 copy --from=main /copy /copy
 run ["upx", "/copy"]
 
-from alpine AS cp
+from alpine AS wget
+workdir /out
 run apk add --no-cache wget
+
+from wget AS cp
 run wget http://s.minos.io/archive/bifrost/x86_64/coreutils-7.6-5.tar.gz
 run tar xvf coreutils-7.6-5.tar.gz -C /
 
-from scratch
-copy --from=upx /copy /bin/copy
-copy --from=cp /bin/cp /bin/cp
-entrypoint ["/bin/copy"]
\ No newline at end of file
+from wget AS tar
+run wget http://s.minos.io/archive/bifrost/x86_64/tar-1.23-1.tar.gz
+run tar xvf tar-1.23-1.tar.gz -C /out
+
+from wget AS gz
+run wget http://s.minos.io/archive/bifrost/x86_64/gzip-1.4-1.tar.bz2 
+run tar xvf gzip-1.4-1.tar.bz2 -C /out
+
+from wget AS bz
+run wget http://s.minos.io/archive/bifrost/x86_64/bzip2-bin-1.0.5-1.tar.gz
+run tar xvf bzip2-bin-1.0.5-1.tar.gz -C /out
+
+from wget AS xz
+run wget http://s.minos.io/archive/bifrost/x86_64/xz-5.0.3-1.tar.gz
+run tar xvf xz-5.0.3-1.tar.gz  -C /out
+
+from scratch AS release
+copy --from=upx /copy /bin/
```

Comment:
Does `upx` work for other binaries as well?

Reply:
I haven't tried.